### PR TITLE
Fix slow op on refreshSubscriptionTab + restart the agent on jetbrains' and agent's user id mismatch

### PR DESCRIPTION
**Pull request by mkondratek**
## Test plan 
- put a breakpoint in `addNewSubscriptionTab` and set `agentUserId` a dummy value so it mismatch the `jetbrainsUserId`(that will test the error handling)
  - internal ide error should not appear 
  - agent should restart and a new proper matching id should be assigned 

- in `addNewSubscriptionTab` increase the timeout & runIde
  - UI should not freeeze 🥶

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 1]
```diff
@@ -111,20 +112,21 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
...
 
     addWelcomeMessage()
-    refreshSubscriptionTab()
+    ApplicationManager.getApplication().executeOnPooledThread { refreshSubscriptionTab() }
     loadNewChatId()
   }
 
+  @RequiresBackgroundThread
   fun refreshSubscriptionTab() {
-    runInEdtAsync(Disposer.newCheckedDisposable()) {
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:119]: tbh, I think we shouldn't have it called on Edt (whatever means "Async" here). for an experiment I switched `Thread.sleep(200)` to wait a few seconds instead - I was freezing the UI 😕 

I marked this method as `RequiresBackgroundThread` and called it on     `ApplicationManager.getApplication().executeOnPooledThread` where we call it.

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 2]
```diff
@@ -111,20 +112,21 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
...
-        if (tabbedPane.tabCount < SUBSCRIPTION_TAB_INDEX + 1) {
-          addNewSubscriptionTab(server)
-        } else {
+    tryRestartingAgentIfNotRunning(project)
+    getInitializedServer(project).thenAccept { server ->
+      if (tabbedPane.tabCount < SUBSCRIPTION_TAB_INDEX + 1) {
+        addNewSubscriptionTab(server)
+      } else {
+        ApplicationManager.getApplication().invokeLater {
           tabbedPane.remove(SUBSCRIPTION_TAB_INDEX)
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:127]: this is an UI actions so we must get back to UI thread (`invokeLater`)

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 3]
```diff
@@ -163,7 +165,9 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
       }
       if (jetbrainsUserId != agentUserId) {
         if (agentUserId != null) {
-          logger.error("User id in JetBrains is different from agent")
+          logger.warn("User id in JetBrains is different from agent: restarting agent...")
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:168]: changed to warn (so a user does not see an internal error anymore) + restarting the agent (which should be a fix)

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 4]
```diff
@@ -181,9 +185,11 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
                   }
                   .get()
           if (isCurrentUserPro != null) {
-            val subscriptionPanel = createSubscriptionTab(isCurrentUserPro)
-            tabbedPane.insertTab(
-                "Subscription", null, subscriptionPanel, null, SUBSCRIPTION_TAB_INDEX)
+            ApplicationManager.getApplication().invokeLater {
+              val subscriptionPanel = createSubscriptionTab(isCurrentUserPro)
+              tabbedPane.insertTab(
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:190]: UI actions - moved to UI thread

File: src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt [hunk 5]
```diff
@@ -111,20 +112,21 @@ class CodyToolWindowContent(private val project: Project) : UpdatableChat {
...
-    runInEdtAsync(Disposer.newCheckedDisposable()) {
-      tryRestartingAgentIfNotRunning(project)
-      getInitializedServer(project).thenAccept { server ->
-        if (tabbedPane.tabCount < SUBSCRIPTION_TAB_INDEX + 1) {
-          addNewSubscriptionTab(server)
-        } else {
+    tryRestartingAgentIfNotRunning(project)
+    getInitializedServer(project).thenAccept { server ->
+      if (tabbedPane.tabCount < SUBSCRIPTION_TAB_INDEX + 1) {
+        addNewSubscriptionTab(server)
```
Comment (pkukielka) [inline src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt:124]: I looked into `addNewSubscriptionTab` and now it seems to be doing both long operations and UI operations.
Now long operations are on the background thread, but UI operations (like `tabbedPane.insertTab`) are not on the dispatch thread.

**Review by Sa1to — APPROVED** [review]
LGTM

**Review by exigow — APPROVED** [review]
Looks good :+1:. I confirm that the agent is forced to restart after ID mismatch. Subscription panel is not freezing.

**Pull request merged**